parquet-column/src/main/java/parquet/filter2/recordlevel/FilteringPrimitiveConverter.java

```diff
@@ -40,23 +40,25 @@ public FilteringPrimitiveConverter(PrimitiveConverter delegate, ValueInspector[]
     this.valueInspectors = checkNotNull(valueInspectors, "valueInspectors");
   }
 
-  // TODO: this works, but
-  // TODO: essentially turns off the benefits of dictionary support
-  // TODO: even if the underlying delegate supports it.
-  // TODO: we should support it here. (https://issues.apache.org/jira/browse/PARQUET-36)
   @Override
   public boolean hasDictionarySupport() {
-    return false;
+    return delegate.hasDictionarySupport();
```

Comment:
Maybe. Its more work for FilteringPrimtive converter, it will need to call different methods on delegates but seems like a sensible idea